src/core/dune

```diff
@@ -1,23 +1,10 @@
 (* -*- tuareg -*- *)
 
 let preprocess =
-  let older =
-    Scanf.sscanf
-      Sys.ocaml_version
-      "%d.%2d.%s"
-      (fun maj min _ -> maj < 4 || (maj = 4 && min < 8))
-  in
-  let bisect =
-    match Sys.getenv "BISECT_ENABLE" with
-    | "yes" -> "(preprocess (pps bisect_ppx))"
-    | _ -> ""
-    | exception _ -> ""
-  in
-  let future_syntax = "(preprocess (future_syntax))" in
-  if older then
-    future_syntax
-  else
-    bisect
+   match Sys.getenv "BISECT_ENABLE" with
+   | "yes" -> "(preprocess (pps bisect_ppx))"
+   | _ -> ""
+   | exception _ -> ""
```

Comment:
I guess with recent dune (>= 2.7.0) and bisect_ppx (2.5.0) this could be done differently; namely in the `dune` files add the stanza `(instrumentation (backend bisect_ppx)))` (as explained in bisect_ppx README). :)